compute_tools/src/http/api.rs

```diff
@@ -165,6 +167,46 @@ async fn routes(req: Request<Body>, compute: &Arc<ComputeNode>) -> Response<Body
             }
         }
 
+        (&Method::POST, "/grants") => {
+            info!("serving /grants POST request");
+            let status = compute.get_status();
+            if status != ComputeStatus::Running {
+                let msg = format!(
+                    "invalid compute status for set_role_grants request: {:?}",
+                    status
+                );
+                error!(msg);
+                return Response::new(Body::from(msg));
```

Comment:
Here, the response is returned as plain text, but in the spec you define it as a GenericError json.

I'd suggest having a look at how `/configure` handles errors
- it returns 412 for the wrong status
- it uses `render_json_error` to render a structured json

It's a pretty small change, but will help us keeping all new APIs consistent (I know that there are some old ones that return plain text)

Reply:
Unrelated, but @lubennikovaav look like `/installed_extensions` is inconsistent in how it returns errors, could be both plain text and json